documentation/stackoverflow.workspace.adoc

```diff
@@ -3,211 +3,336 @@
+* How are users related to each other?
+* Which questions are unanswered?
+* What answers are the most popular?
+
+The following query will use the `ASKED` relationship to find `User` and `Question` nodes.
+
+.Match User and Question nodes
 [source,cypher]
 ----
-CALL db.schema.visualization();
+MATCH (u:User)-[a:ASKED]->(q:Question)
+RETURN u,a,q
 ----
 
-== Load JSON Import
+[NOTE]
+====
+The duplicate button icon:ArrowIcon[] will copy the query to the clipboard.
+
+The play button icon:PlayIcon[] will execute the query and return the results.
+====
 
-There is already data preloaded in this graph, if you want to extend (more pages or other tags) or update it with the most recent questions, please modify and run the statement below.
+Run the query and observe the results. You should note that the `ASKED` relationship is in the direction `User` to `Question`.
 
-Update this dataset using `apoc.load.json`.
- Run the query as-is to update or add a tag to the `tags` array to extend. 
+[NOTE]
+.Challenge
+====
+Can you complete this query to find `User` and `Answer` nodes using the `PROVIDED` relationship:
 
 [source,cypher]
 ----
-// look for several pages of questions
-WITH ["neo4j","cypher"] as tags 
-UNWIND tags as tagName
-UNWIND range(1,2) as page
-
-WITH "https://api.stackexchange.com/2.3/questions?page="+page+"&pagesize=25&order=desc&sort=creation&tagged="+tagName+"&site=stackoverflow&filter=!5-i6Zw8Y)4W7vpy91PMYsKM-k9yzEsSC1_Uxlf" as url
-
-CALL apoc.load.json(url) YIELD value
-CALL apoc.util.sleep(250)  // careful with throttling
-
-UNWIND value.items AS q
-
-// create the questions
-MERGE (question:Question {uuid:q.question_id})
-  ON CREATE SET question.title = q.title, 
-  	question.link = q.share_link, 
-  	question.creation_date = q.creation_date, 
-  	question.accepted_answer_id=q.accepted_answer_id, 
-  	question.view_count=q.view_count,
-   	question.answer_count=q.answer_count,
-   	question.body_markdown=q.body_markdown
-
-// who asked the question
-MERGE (owner:User {uuid:coalesce(q.owner.user_id,'deleted')})
-  ON CREATE SET owner.display_name = q.owner.display_name
-MERGE (owner)-[:ASKED]->(question)
-
-// what tags do the questions have
-FOREACH (tagName IN q.tags | 
-  MERGE (tag:Tag {name:tagName}) 
-    ON CREATE SET tag.link = "https://stackoverflow.com/questions/tagged/" + tag.name
-  MERGE (question)-[:TAGGED]->(tag))
-
-// who answered the questions?
-FOREACH (a IN q.answers |
-   MERGE (question)<-[:ANSWERED]-(answer:Answer {uuid:a.answer_id})
-    ON CREATE SET answer.is_accepted = a.is_accepted,
-    answer.link=a.share_link,
-    answer.title=a.title,
-    answer.body_markdown=a.body_markdown,
-    answer.score=a.score,
-   	answer.favorite_score=a.favorite_score,
-   	answer.view_count=a.view_count
-   MERGE (answerer:User {uuid:coalesce(a.owner.user_id,'deleted')}) 
-    ON CREATE SET answerer.display_name = a.owner.display_name
-   MERGE (answer)<-[:PROVIDED]-(answerer)
-)
+MATCH (u:User)-[ ]->( )
```

Comment:
do you want to leave it like this (which is nice because it's executable)

or put in a place holder like `MATCH (u:User)-[/* fill in relationship-type */]->( /* node label here */)` ?

Reply:
I think giving the user a hint is s good idea. I have added ? marks as placeholders.